cajita/src/Cajita_SparseIndexSpace.hpp

```diff
@@ -0,0 +1,564 @@
+#ifndef CAJITA_SPARSE_INDEXSPACE_HPP
+#define CAJITA_SPARSE_INDEXSPACE_HPP
+
+#include <Kokkos_Atomic.hpp>
+#include <Kokkos_Core.hpp>
+#include <Kokkos_UnorderedMap.hpp>
+
+#ifdef __CUDACC__
+#define CUDA_HOSTDEV __host__ __device__
```

Comment:
Kokkos has macros for this already. Delete this and just use `KOKKOS_INLINE_FUNCTION` everywhere that it might be a host/device function if the GPU is used.